IdentityCore/src/broker_operation/response/MSIDBrokerOperationResponse.m

```diff
@@ -33,17 +38,26 @@ - (instancetype)initWithJSONDictionary:(NSDictionary *)json error:(NSError **)er
     
     if (self)
     {
-        // TODO: implement.
+        if (![json msidAssertType:NSString.class ofKey:MSID_BROKER_OPERATION_JSON_KEY required:YES error:error]) return nil;
+        _operation = json[MSID_BROKER_OPERATION_JSON_KEY];
+
+        _success = [json[MSID_BROKER_OPERATION_RESULT_JSON_KEY] boolValue];
+        
+        _deviceInfo = [[MSIDDeviceInfo alloc] initWithJSONDictionary:json error:error];
     }
     
     return self;
 }
 
 - (NSDictionary *)jsonDictionary
 {
-    // TODO: implement.
+    NSMutableDictionary *json = [NSMutableDictionary new];
+    json[MSID_BROKER_OPERATION_JSON_KEY] = self.operation;
+    json[MSID_BROKER_OPERATION_RESULT_JSON_KEY] = [@(self.success) stringValue];
+    
+    json[@"device_info"] = self.deviceInfo.jsonDictionary;
```

Comment:
this creates a nested dictionary, but initializer doesn't expect a nested dictionary.

Reply:
addressed :)